tools/hilbert.py

```diff
@@ -289,23 +289,91 @@ def cmd_list_applications(parser, context, args):
     group.add_argument('--configdump', required=False,
                        help="specify input dump file")
 
+    parser.add_argument('-f', '--format', choices=['list', 'dashboard'],
+                        default='list', required=False,
+                        help="output format")
+
     args = parser.parse_args(args)
 
-    log.debug("Listing all Application ID...")
+    _args = vars(args)
+
+    mode = _args.get('format', 'list')
+
+    if mode == "list":
+        log.debug("Listing all Application ID...")
+
+        obj = None
+        try:
+            obj = cmd_list(parser, context, args, 'Applications/keys')
+        except:
+            log.exception("Sorry could not get the list of '{}' from the input file!".format('applications'))
+            sys.exit(1)
+
+        assert obj is not None
+
+        if isinstance(obj, BaseValidator):
+            print(yaml_dump(obj.data_dump()))
+        else:
+            print(yaml_dump(obj))
+
+        return args
+
```

Comment:
it depends, for instance multiple if-branches with long code blocks may make the logic harder to understand. Not sure about this place, but i usually prefer to return as soon as possible if that makes logic easier to see...

Reply:
If the code block in an `if` is long and hard to understand, factor it out into a separate (private) function that has a well chosen name. Then, you only have 
```Python
if condition1:
    function1(...)
elif condition2:
    function2(...)
else:
   # this should never happen -> error handling
```
Which makes it trivial, to see what happens. Even
```Python
if condition1:
    function1(...)
    return
elif condition2:
    function2(...)
    return
else:
   # this should never happen -> error handling
```
is more clear then
```Python
if condition1:
    function1(...)
    return

assert condition2
function2(...)
```
because you have to understand what happens in the first `if` to understand the control flow.

There is one case, where early `return` usually simplifies code and this is if you are checking the precondition of a function:
```Python
if(err)
    return

# actually do something
```
Which in some cases can also be written as
```Python
assert not err, "Useful error message" 

# actually do something
```
I am not going to enforce anything like this because the discussion about such things can go on forever. Nevertheless I'll comment on code that IMO contains bad pratices.